lib/backend/fs.dart

```diff
@@ -0,0 +1,513 @@
+    );
+  }
+
+  TransferFileOperation copy({required File to, Cancellable? cancellable}) {
+    return TransferFileOperation._(
+      source: this,
+      destination: to,
+      copy: true,
+      cancellable: cancellable,
+    );
+  }
+
+  FileSystemOperation<bool> delete({Cancellable? cancellable}) {
+    return FileSystemOperation<bool>._(
+      target: _handle.cast(),
+      onCreateCompleteCallable: NativeCallable<Void Function(Bool)>.listener,
+      onStartOperation: (cancellable, onComplete, error) =>
+          file_delete(_handle, cancellable, onComplete.cast(), error),
+      cancellable: cancellable,
+    );
+  }
+
+  FileSystemOperation<FileEnumerator> getEnumerator({
+    String attributes = '*',
+    Cancellable? cancellable,
+  }) {
+    return FileSystemOperation<FileEnumerator>._(
+      target: _handle.cast(),
+      onCreateCompleteCallable: (onComplete) =>
+          NativeCallable<Void Function(Pointer<GFileEnumerator>)>.listener(
+            (Pointer<GFileEnumerator> v) => onComplete(FileEnumerator._(v)),
+          ),
+      onStartOperation: (cancellable, onComplete, error) =>
+          file_enumerate_children(
+            _handle,
+            attributes.toNativeUtf8().cast(),
+            0,
+            cancellable,
+            onComplete.cast(),
+            error,
+          ),
+      cancellable: cancellable,
+    );
+  }
+
+  TransferFileOperation move({required File to, Cancellable? cancellable}) {
+    return TransferFileOperation._(
+      source: this,
+      destination: to,
+      copy: false,
+      cancellable: cancellable,
+    );
+  }
+
+  FileSystemOperation<FileInfo> queryInfo({
+    String attributes = '*',
+    Cancellable? cancellable,
+  }) {
+    return FileSystemOperation<FileInfo>._(
+      target: _handle.cast(),
+      onCreateCompleteCallable: (onComplete) =>
+          NativeCallable<Void Function(Pointer<GFileInfo>)>.listener(
+            (Pointer<GFileInfo> v) => onComplete(FileInfo._(v)),
+          ),
+      onStartOperation: (cancellable, onComplete, error) => file_query_info(
+        _handle,
+        attributes.toNativeUtf8().cast(),
+        0,
+        cancellable,
+        onComplete.cast(),
+        error,
+      ),
+      cancellable: cancellable,
+    );
+  }
+
+  FileSystemOperation<bool> trash({Cancellable? cancellable}) {
+    return FileSystemOperation<bool>._(
+      target: _handle.cast(),
+      onCreateCompleteCallable: (onComplete) =>
+          NativeCallable<Void Function(Bool)>.listener(onComplete),
+      onStartOperation: (cancellable, onComplete, error) =>
+          file_trash(_handle, cancellable, onComplete.cast(), error),
+      cancellable: cancellable,
+    );
+  }
+}
+
+extension type FileInfo._(Pointer<GFileInfo> _handle) {
+  String? getName() {
+    final name = fileinfo_get_name(_handle);
+    if (name == nullptr) return null;
+
+    try {
+      return name.cast<Utf8>().toDartString();
+    } finally {
+      calloc.free(name);
+    }
```

Comment:
https://docs.gtk.org/gio/method.FileInfo.get_name.html#return-value
this should not get freed here, the data is owned by the file info instance